fdm-calculator/src/balance/nitrogen/index.test.ts

```diff
@@ -0,0 +1,196 @@
+import { describe, expect, it } from "vitest"
+import {
+    calculateNitrogenBalance,
+    combineSoilAnalyses,
+    getFdmPublicDataUrl,
+} from "."
+import type {
+    FieldInput,
+    NitrogenBalanceInput,
+    SoilAnalysisPicked,
+} from "./types"
+
+describe("getFdmPublicDataUrl", () => {
+    it("should return the correct FDM public data URL", () => {
+        const expectedUrl = "https://storage.googleapis.com/fdm-public-data/"
+        expect(getFdmPublicDataUrl()).toBe(expectedUrl)
+    })
+})
+
+describe("calculateNitrogenBalance", () => {
+    it("should calculate nitrogen balance correctly with mock input", async () => {
+        const mockNitrogenBalanceInput: NitrogenBalanceInput = {
+            fields: [
+                {
+                    field: {
+                        b_id: "field1",
+                        b_centroid: [5.0, 52.0],
+                        b_area: 100,
+                        b_start: new Date("2023-01-01"),
+                        b_end: new Date("2023-12-31"),
+                    },
+                    cultivations: [
+                        {
+                            b_lu: "cultivation1",
+                            b_lu_catalogue: "catalogue1",
+                            m_cropresidue: true,
+                        },
+                    ],
+                    harvests: [
+                        {
+                            b_id_harvesting: "harvest1",
+                            b_lu: "cultivation1",
+                            b_lu_harvest_date: new Date(),
+                            harvestable: {
+                                b_id_harvestable: "harvestable1",
+                                harvestable_analyses: [
+                                    {
+                                        b_lu_yield: 1000,
+                                        b_lu_n_harvestable: 20,
+                                    },
+                                ],
+                            },
+                        },
+                    ],
+                    soilAnalyses: [
+                        {
+                            a_id: "soil1",
+                            b_sampling_date: new Date(),
+                            a_c_of: 20,
+                            a_cn_fr: 10,
+                            a_density_sa: 1.2,
+                            a_n_rt: 1,
+                            a_som_loi: 2,
```

Comment:
currently values are not evaluated whether they are correct? when a_c_of is 20 and CN is 10, then a_n_rt is not 1. might not be a problem here.